pkg/docker/config/config.go

```diff
@@ -587,6 +607,16 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 		return getAuthFromCredHelper(ch, registry)
 	}
 
+	// Support for different paths in auth.
+	if ref != nil {
+		return findAuthenticationForRef(auths, ref)
```

Comment:
Non-blocking: Is this worth the extra two functions?

If the goal is testing, splitting this into “read file+parse JSON” and “interpret in-memory data”, or maybe “read file” + “parse and interpret byte stream” would allow better testing, because that would test the prioritization of repo/registry credentials.

Reply:
No, I just refactored them back into the original function.